roles/ceph-defaults/defaults/main.yml

```
@@ -338,10 +338,12 @@ cephfs_pools:
 
 ## OSD options
 #
+# Variables to calculate a suitable osd_memory_target based on the number of OSDs
 is_hci: false
 hci_safety_factor: 0.2
 non_hci_safety_factor: 0.7
-osd_memory_target: 4294967296
+# You can also set an explicit value which will override the calculation if set.
+# osd_memory_target: 4294967296
```

This should be left uncommented out
```suggestion
osd_memory_target: 4294967296
```

otherwise the `osd memory target = {{ _osd_memory_target | default(osd_memory_target) }}` in ceph.conf.j2 doesn't make sense anymore since it won't work

---

Your proposed fix won't work. If the `osd_memory_target` is defined in the defaults, it will always take precedence after my change. AFAIK you can't unset a variable that's set in role defaults and the logic is based on the assumption that the calculated value should be used if `osd_memory_target` is not defined.

But there is indeed an issue if `_num_osds == 0` because then `_osd_memory_target` will be undefined even if `osd_memory_target` is not defined. I fixed this by adding another default value directly to the template to cover this edge case .